master/getting-started/bare-metal/bare-metal.md

```diff
@@ -427,3 +430,284 @@ an ingress rule allowing access *to* port 999 and an egress rule allowing
+additional label to indicate that, let's say `load-balancer-ingress: true`.
+
+Then we can deny all incoming traffic through those interfaces, unless it is
+from a source IP that is known to be within the cluster.  (Note: we are
+assuming that the same interfaces can also be used for traffic that is
+forwarded from other nodes or pods in the cluster - as would be the case for
+nodes with only one external interface.)
+
+```
+apiVersion: v1
+kind: policy
+metadata:
+  name: disallow-incoming
+spec:
+  preDNAT: true
+  order: 100
+  ingress:
+    - action: deny
+      source:
+        notNets: [<pod-cidr>, <cluster-internal-node-cidr>, ...]
+  selector: host-endpoint=='ingress'
+```
+
+Now, to allow traffic through the load balancer ingress nodes to
+`<service-load-balancer-ip>:<service-port>` (for each load-balanced Service):
+
+```
+apiVersion: v1
+kind: policy
+metadata:
+  name: allow-load-balancer-service-1
+spec:
+  preDNAT: true
+  order: 90
+  ingress:
+    - action: allow
+      destination:
+        nets: [<service-load-balancer-ip>]
+        ports: [<service-port>]
+  selector: load-balancer-ingress=='true'
+```
+
+And for traffic to NodePorts - for each non-load-balanced Service - via any
+node:
+
+```
+apiVersion: v1
+kind: policy
+metadata:
+  name: allow-node-port-service-1
+spec:
+  preDNAT: true
+  order: 90
+  ingress:
+    - action: allow
+      destination:
+        ports: [<node-port>]
+  selector: host-endpoint=='ingress'
+```
+
+And that completes the example.  It's worth re-emphasizing, though, two key
+points about the application of pre-DNAT policy that make this work; especially
+as pre-DNAT policy differs on these points from normal host endpoint policy.
+
+Firstly, there is no 'default drop' semantic for pre-DNAT policy, like there
+_is_ for normal policy.  So, if policies are defined such that _some_ pre-DNAT
+policies apply to a host endpoint, but none of those policies matches a
+particular incoming packet, that packet is allowed to continue on its way.
+(Whereas if there are normal policies that apply to a host endpoint, and
+none of those policies matches a packet, that packet will be dropped.)
+
+For the example here, that means that we can specify some pre-DNAT policy,
+applying to all of the cluster's external interfaces, without having to
+enumerate and explicitly _allow_ all of the internal flows that may also go
+through those interfaces.  It's also why the second point works...
+
+Namely, that if traffic comes in through a host endpoint and is routed to a
+local workload, any host endpoint pre-DNAT policy is enforced as well as the
+ingress policy for that workload - whereas normal host endpoint policy is
+skipped in that scenario.  (Normal host endpoint policy is 'trumped' by
+workload policy, for packets going to a local workload.)
+
+For the example here, that means that the last pre-DNAT policy above does not
+accidentally expose workloads that happen to use the same `<node-port>`, or
```

Comment:
Yes, the behavior Neil describes is what we want (and is hopefully what he's implemented too).